File: 1897/solution.py
from collections import defaultdict


class Solution:
    def makeEqual(self, words: list[str]) -> bool:
        counter = defaultdict(int)
        W = len(words)
        for word in words:
            for char in word:
                counter[char] += 1

        for val in counter.values():
            if val % W:
                return False
        return True
        # time O(c * n)
        # space O(1)

File: 1897/solution.ts
function makeEqual(words: string[]): boolean {
  const counts = new Array(26).fill(0);
  for (const word of words) {
    for (const char of word) {
      counts[char.charCodeAt(0) - "a".charCodeAt(0)]++;
    }
  }
  for (const num of counts) {
    if (num % words.length != 0) {
      return false;
    }
  }
  return true;
}
